### PR TITLE
Add guest mode

### PR DESCRIPTION
**Pull request by davilajose23**
## Description
This PR adds the possibility to see the sessions without an account.

## How has this been tested?

Please mark the tests that you ran to verify your changes. If difficult to test, consider providing instructions so reviewers can test.

- [ ] Manual testing
- [ ] System tests
- [ ] Unit tests
- [ ] None

## Checklist

- [ ] CI pipeline is passing
- [ ] My code follows the conventions of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (if applicable)
- [ ] I have added seed data to the database (if applicable)

## Release tasks

Add any tasks that need to be done before/after the release of this feature.

## Screenshots/Loom
image

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,5 +1,11 @@
 Rails.application.routes.draw do
-  root "sessions#index"
+  constraints AuthenticatedConstraint.new do
+    root "sessions#index"
+  end
+
+  constraints UnauthenticatedConstraint.new do
+    root "user_sessions#new", as: :unauthenticated_root
+  end
```
Comment (davilajose23) [inline config/routes.rb:8]: Constraints allow us to have two different root paths. https://api.rubyonrails.org/classes/ActionDispatch/Routing/Mapper/Scoping.html#method-i-constraints

File: app/views/user_sessions/new.html.erb [hunk 2]
```diff
@@ -22,7 +22,8 @@
 
         <div class="flex flex-col items-center mb-10">
           <%= link_to "Forgot password?", new_password_reset_path, class: "font-black italic underline text-white mb-10" %>
-          <%= link_to "Sign up", new_registration_path, class: "font-black italic underline text-white" %>
+          <%= link_to "Sign up", new_registration_path, class: "font-black italic underline text-white mb-10" %>
+          <%= link_to "Continue as guest", sessions_path, class: "font-black italic underline text-white" %>
```
Comment (davilajose23) [inline app/views/user_sessions/new.html.erb:26]: I am waiting for Charly's opinion on the design.

Comment (davilajose23) [reply app/views/user_sessions/new.html.erb]: I updated the PR based on this design:
image

File: app/controllers/abouts_controller.rb [hunk 3]
```diff
@@ -1,4 +1,5 @@
 class AboutsController < ApplicationController
+  skip_before_action :authenticate_user!
```
Comment (andresag4) [inline app/controllers/abouts_controller.rb:2]:
```suggestion
  allow_unauthenticated_access
```
We have this helper method in Authentication module.

File: app/controllers/sessions_controller.rb [hunk 4]
```diff
@@ -1,6 +1,8 @@
 class SessionsController < ApplicationController
+  skip_before_action :authenticate_user!
```
Comment (andresag4) [inline app/controllers/sessions_controller.rb:2]:
```suggestion
  allow_unauthenticated_access
```

File: app/models/session.rb [hunk 5]
```diff
@@ -49,6 +50,8 @@ class Session < ApplicationRecord
   scope :starting_soon, -> { where("starts_at BETWEEN ? and ?", Time.current, 1.hour.from_now) }
   scope :upcoming_today, -> { where("date(starts_at) = ? and starts_at > ?", Date.current, Time.current) }
   scope :live_or_upcoming_today, -> { live.or(upcoming_today) }
+  scope :publics, -> { where(public: true) }
+  scope :privates, -> { where(public: false) }
```
Comment (andresag4) [inline app/models/session.rb:54]:
```suggestion
  scope :public, -> { where(public: true) }
  scope :private, -> { where(public: false) }
```

would opt for singular naming scopes

File: app/queries/session_query.rb [hunk 6]
```diff
@@ -27,6 +28,12 @@ def filter_by_status
     self.relation = relation.send_chain_or(status_scopes)
   end
 
+  def filter_privates
+    return if params["show_private"].present?
```
Comment (andresag4) [inline app/queries/session_query.rb:32]:
```suggestion
    return if params[:show_private].present?
```

File: app/controllers/speakers_controller.rb [hunk 7]
```diff
@@ -1,4 +1,5 @@
 class SpeakersController < ApplicationController
+  skip_before_action :authenticate_user!, only: [:show]
```
Comment (andresag4) [inline app/controllers/speakers_controller.rb:2]:
```suggestion
  allow_unauthenticated_access only: [:show]
```

File: app/views/layouts/_bottom_navbar.html.erb [hunk 8]
```diff
@@ -18,7 +18,7 @@
       <%= link_to(
         schedule_path(
           starts_at: current_starts_at_filter,
-          anchor: session_anchor(current_user.sessions.live_or_upcoming_today.first)
+          anchor: session_anchor(current_user&.sessions&.live_or_upcoming_today&.first)
```
Comment (andresag4) [inline app/views/layouts/_bottom_navbar.html.erb:21]:
```suggestion
          anchor: session_anchor(current_schedule_session)
```

and add the safe operator on the session_helper.rb

```
  def current_schedule_session
    @current_schedule_session ||= current_user&.sessions&.live_or_upcoming_today&.first
  end
```

File: app/views/layouts/_bottom_navbar.html.erb [hunk 9]
```diff
@@ -29,18 +29,34 @@
...
-        ]
-      ) do %>
-        <%= inline_svg_tag(
-          "icons/avatar_no_fill.svg",
-          class: nav_icon_class_for(["/profiles"])
-        ) %>
-        Profile
+      <% if user_signed_in? %>
+        <%= link_to(
+          profile_path(current_profile&.uuid),
```
Comment (andresag4) [inline app/views/layouts/_bottom_navbar.html.erb:34]: Seems that only the path changes for the if/else

```suggestion
        user_signed_in? ? profile_path(current_profile&.uuid) : new_user_session_path
```

**Review by andresag4 — CHANGES_REQUESTED** [review]

File: app/helpers/navigation_helper.rb [hunk 10]
```diff
@@ -22,8 +22,14 @@ def show_header?
...
-    user_signed_in? &&
-      !current_page?(coming_soon_path)
+    current_page?(sessions_path) ||
+      (user_signed_in? || !current_page?(unauthenticated_root_path)) &&
+        !current_page?(new_registration_path) &&
+        !current_page?(new_user_session_path) &&
+        !current_page?(new_password_reset_path) &&
+        !current_page?(edit_password_reset_path) &&
+        !current_page?(post_submit_password_reset_path) &&
+        !current_page?(coming_soon_path)
```
Comment (andresag4) [inline app/helpers/navigation_helper.rb:32]:
```suggestion
  def show_bottom_navbar?
    current_page?(sessions_path) ||
      (user_signed_in? || !current_page?(unauthenticated_root_path)) &&
        !excluded_paths.any? { |path| current_page?(path) }
  end
  
  private
  
  def excluded_paths
    [
      new_user_session_path,
      new_registration_path,
      new_password_reset_path,
      edit_password_reset_path,
      post_submit_password_reset_path
    ]
  end
```

**Review by andresag4 — CHANGES_REQUESTED** [review]

File: app/models/session.rb (hunk 5 above)
Comment (davilajose23) [reply app/models/session.rb:54]: I decided to use plurar scopes because singular words`private` and `public`  are reserved.

Comment (andresag4) [reply app/models/session.rb:54]: yep, just realized that

File: app/controllers/sessions_controller.rb [hunk 11]
```diff
@@ -9,6 +11,7 @@ def index
 
   def show
     @session = sessions.friendly.find(params[:id])
+    raise ActiveRecord::RecordNotFound if @session.private? && !user_signed_in?
```
Comment (andresag4) [inline app/controllers/sessions_controller.rb:14]: another option would be this:
```suggestion
    if user_signed_in?
      @session = sessions.friendly.find(params[:id])
    else
      @session = sessions.publics.friendly.find(params[:id])
    end
```

**Pull request merged**